alb/alb.go

```diff
@@ -106,7 +110,7 @@ func (a *alb) Stop() error {
 
 // Health returns nil if attached to all frontends.
 func (a *alb) Health() error {
-	if len(a.targetGroupNames) == a.registeredFrontends {
+	if !a.readyForHealthCheck.Get() || len(a.targetGroupNames) == a.registeredFrontends {
```

Comment:
I believe the usage of registeredFrontends needs to be guarded by a mutex. Since `#Health` will be called in a separate goroutine from `#Update`. `targetGroupNames` is okay because it is initialised before the goroutines are started.